### ISSUE TITLE
API Usage Error for enc_dec generate with hybrid Swin encoder (trt) and Bart decoder (trt_llm)

### ISSUE THREAD

**Issue by jpototpbys** [NVIDIA/TensorRT-LLM#3620]
I trying to convert Donut model (Swin encoder and Bart decoder) using trt for the encoder (which is working properly) and trt_llm for the decoder (using template in examples enc_dec).

The conversion using the script python found in example enc_dec + the script trtllm-build   works to convert from torch the decoder to trt_llm engine.

Unfortunately, I have some issues when I want to infer the decoder using generate (and activated skip_encoder=True) or using decoder_session.decode. I got the following errors: 

![Image](https://github.com/user-attachments/assets/bd246910-ba9d-42cb-9838-1ddb5575e3cc)

My encoder takes an image as input of size 1x3x2560x1920. It gives an embedding of shape 4800x1024

My decoder (in onnx or torch) takes as input the embeddings of shape 1x4800x1024 and the prompt tokenized of shape 1x1
It gives as output a logits tensor of shape 1024x57541 (before decoding with tokenizer).

I set as follow the generate params:
input_ids: tensor of shape [1] (I try also with the image input of the encoder of shape 1x3x2560x1920)
decoder_input_ids: tensor of shape [1] 
prompt_embedding_table: output encoder tensor of shape [4800,1024]
max_new_tokens: 1024
return_dict: True
time_encoder: False
return_encoder_output: False

**Comment by karljang** [comment]
Sorry we never replied to this one. Closing as stale since the TensorRT engine-build path is deprecated in favor of the PyTorch backend. Please open a new issue if the problem persists in the latest version.
